### ISSUE TITLE
VSCode wants to change my file's permissions (back) to 644

### ISSUE THREAD

**Issue by rudiedirkx** [Microsoft/vscode#41400]
- VSCode Version: 1.19.1
- OS Version: Win 7

I've just installed VSCode for the very first time. No plugins added or preferences changed.

I'm loading a project over LAN (Samba). I'm on Windows, the project is on Linux. VSCode GIT sees a few files as Untracked, which makes sense, because they're only in the global gitignore on Linux, so VSCode doesn't know that. But there is an executable (775 on Linux, but GIT sees that as 755) that VSCode says is a CHANGE. `Open changes` doesn't highlight anything. Linux GIT doesn't highlight anything. When I `Discard changes` (takes 10 sec!?), the file is still in the CHANGES list, and `git diff` on Linux says:

```
old mode 100755
new mode 100644
```

Long story short: VSCode wants the file to be 644, and even then still sees the file as CHANGED.

Another, new scenario: Linux GIT sees the file as unchanged, VSCode as changed. Now I tell VSCode to `Stage changes` (takes only 0.5 sec). The UI says the file is a STAGED CHANGE. When I look in Linux GIT:

```
Changes to be committed:
        modified:   artisan

Changes not staged for commit:
        modified:   artisan
```

The staged diff:

```
old mode 100755
new mode 100644
```

and the unstaged diff:

```
old mode 100644
new mode 100755
```

Go home VSCode, you're drunk!

**Comment by rudiedirkx** [comment]
Ah yes, I see. The diff on Windows says 

```
old mode 100755
new mode 100644
```

PhpStorm doesn't see that difference. VSCode does. Maybe because Windows doesn't really have executable, PS ignores it.

Is there a way to ssh tunnel for git? I've no idea how Samba works. I'd rather trust Linux' GIT, and it's much faster for big changes.

**Comment by joaomoreno** [comment]
Unfortunately there is no current way to do this. My suggestion is to clone the repository in Windows and work directly there.

**Comment by rudiedirkx** [comment]
Or fix Samba =) Thanks
